product-service/src/functions/createProduct/handler.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  APIGatewayProxyEventHandler,
+  formatBadRequestResponse,
+  formatJSONResponse,
+  formatServerErrorResponse
+} from '@libs/api-gateway';
+import { randomUUID } from 'crypto';
+import { addProductToDB, addStockToDB, getProducts } from '@libs/data-service';
+import { CreateProductParams } from 'src/types/api-types';
+
+const validateParams = (data: CreateProductParams) => {
+  const requiredFields: (keyof CreateProductParams)[] = ['title', 'description', 'image', 'price'];
+  const missingFields = requiredFields.filter((field) => !(field in data));
+
+  if (missingFields.length > 0) {
+    const errorFields = missingFields.join(', ');
+
+    const error = `Error: ${errorFields} ${missingFields.length > 1 ? 'are' : 'is'} not provided`;
+
+    return error;
+  } else {
+    return null;
+  }
+};
+
+export const main: APIGatewayProxyEventHandler = async (event) => {
+  try {
+    const productId = randomUUID();
+
+    const data = JSON.parse(event.body || '{}') as unknown as CreateProductParams;
+
+    console.log('Request: create product', { data });
+
+    const validationError = validateParams(data);
```

Comment:
**Question**: Does schema validation  not work  for you from the box?

Reply:
I didn't use it from the scratch and didn't have time to investigate how it works out of the box I'm afraid, but maybe I could spare some time checking that later

Reply:
I played around with middy some hours and it quite handy though at the end I didn't make it to make @middy/http-cors and @middy/validator friends... so my lambda worked only in test mode in AWS but not in browser and postman. I'll leave it for the future cuz want to figure it out if you don't mind